rs/node_rewards/rewards_calculation/src/performance_based_algorithm/v1/e2e_tests.rs

```diff
@@ -896,54 +781,60 @@ fn test_zero_blocks_edge_cases() {
                     node_id: test_node_id(3),
                     node_reward_type: NodeRewardType::Type1,
+                    region: "North America,USA,California".into(),
+                    dc_id: "dc3".into(),
+                },
+                RewardableNode {
+                    node_id: test_node_id(4),
+                    node_reward_type: NodeRewardType::Type1,
                     region: "Europe,Switzerland".into(),
                     dc_id: "dc3".into(),
                 },
             ],
         );
 
-    let result =
-        RewardsCalculationV1::calculate_rewards(&day, &day, Some(provider_id), data_provider)
-            .expect("Calculation should succeed");
+    let result = RewardsCalculationV1::calculate_rewards(day, day, fake_input_provider)
+        .expect("Calculation should succeed");
 
     let daily_result = &result.daily_results[&day];
     let provider_result = &daily_result.provider_results[&provider_id];
 
-    // Subnet failure rate should be 1.0 (75th percentile of 0, 1.0, 0)
-    let subnet_fr = daily_result.subnets_fr[&subnet_id];
-    assert_eq!(subnet_fr, dec!(1.0));
-
-    let node_results = &provider_result.nodes_results;
-    assert_eq!(node_results.len(), 3);
-
-    // Node 1: 0% failure rate, relative FR = max(0, 0 - 1.0) = 0 (no penalty)
-    let node1 = node_results
-        .iter()
-        .find(|n| n.node_id == test_node_id(1))
-        .unwrap();
-    assert_eq!(node1.performance_multiplier, dec!(1.0));
-    assert_eq!(node1.rewards_reduction, dec!(0.0));
-
-    // Node 2: 100% failure rate, relative FR = max(0, 1.0 - 1.0) = 0 (no penalty)
-    let node2 = node_results
-        .iter()
-        .find(|n| n.node_id == test_node_id(2))
-        .unwrap();
-    assert_eq!(node2.performance_multiplier, dec!(1.0));
-    assert_eq!(node2.rewards_reduction, dec!(0.0));
+    // Subnet failure rate should be 1.0 (75th percentile of 0, 0, 0, 1.0)
+    let subnet_fr = daily_result.subnets_failure_rate_percent[&subnet_id];
+    assert_eq!(subnet_fr, dec!(0));
 
-    // Node 3: 0% failure rate, relative FR = max(0, 0 - 1.0) = 0 (no penalty)
-    let node3 = node_results
+    // Convert node_results into a HashMap for direct access by node_id
+    let node_results_map: HashMap<_, _> = provider_result
+        .daily_nodes_rewards
         .iter()
-        .find(|n| n.node_id == test_node_id(3))
-        .unwrap();
-    assert_eq!(node3.performance_multiplier, dec!(1.0));
-    assert_eq!(node3.rewards_reduction, dec!(0.0));
+        .map(|n| (n.node_id, n))
+        .collect();
+
+    // Node 1
+    let node1 = node_results_map.get(&test_node_id(1)).unwrap();
+    if let DailyNodeFailureRate::SubnetMember { node_metrics } = &node1.daily_node_failure_rate {
+        assert_eq!(node_metrics.original_failure_rate_percent, dec!(0));
+        assert_eq!(node_metrics.relative_failure_rate_percent, dec!(0));
+    } else {
+        panic!("Node 1 is not a SubnetMember");
+    }
```

Comment:
Consider doing these like so:

```suggestion
    let node_metrics = match &node1.daily_node_failure_rate {
        DailyNodeFailureRate::SubnetMember { node_metrics } => node_metrics,
        wrong => panic!("{wrong:?}"),
    };
    assert_eq!(node_metrics.original_failure_rate_percent, dec!(0));
    assert_eq!(node_metrics.relative_failure_rate_percent, dec!(0));
```

I would say `if let` should only be used when the enum is Option, like so:

```
if let Some(foo) = foo {
    whatever(foo);
} else {
    handle_none();
}
```

But even then, I think the following is usually better:

```
let Some(foo) = foo else {
    handle_none();
    return;
}

whatever(foo);
```

(This assumes that returning right after `handle_none()` is the appropriate thing to do when `foo` is `None`, which is very often the case.)